File: packages/compiler/playground/ios/Diez/Sources/File/File.swift
import Foundation

public class File : NSObject, Codable {
    var src: String

    public init(withSrc src: String) {
        self.src = src
        super.init()
    }

    public func url() -> URL? {
        // TODO: when we are not in development, we should load the file from a local bundle URL.
        // This will look something like: Bundle.main.url(forResource: "index", withExtension: "html")
        // except we will be loading from the framework bundle (not main). Probably this should be handled
        // inside class Environment {...}.
        let fqUrl = environment.isDevelopment
            ? "\(environment.serverUrl)\(src)"
            : "TODO"

        return URL(string: fqUrl)
    }
}

File: examples/playground/android/diez/src/main/java/ai/haiku/diez/file/File.kt
package ai.haiku.diez.file

import ai.haiku.diez.puente.Environment
import android.net.Uri
import java.net.URL

data class File(val src: String) {
    fun canonicalURL(): String {
        if (Environment.isDevelopment) {
            return "${Environment.serverUrl}${src}"
        }

        // TODO: when we are not in development, we should load the file from a local bundle URL.
        // This will probably look something like: file:///android_asset/diez/path/to/asset.extension
        return "TODO"
    }

    fun uri(): Uri {
        return Uri.parse(canonicalURL())
    }

    fun url(): URL {
        return URL(canonicalURL())
    }
}
